src/formatDate.js

```diff
@@ -50,7 +50,47 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  // if (fromFormat[0] === 'YYYY' && toFormat[0] === 'YYYY') {
+  //   return date.split(fromFormat[3]).join(toFormat[3]);
+  // }
+
+  // if (fromFormat[0] === 'YYYY' && toFormat[0] === 'DD') {
+  //   return date.split(fromFormat[3]).reverse().join(toFormat[3]);
+  // }
+
+  const [data1, data2, data3] = date.split(fromFormat[3]);
+
+  fromFormat[0] = `${fromFormat[0]}` + `${data1}`;
+  fromFormat[1] = `${fromFormat[1]}` + `${data2}`;
+  fromFormat[2] = `${fromFormat[2]}` + `${data3}`;
+
+  for (let i = 0; i < 3; i++) {
+    for (let y = 0; y < 3; y++) {
+      if (toFormat[i] === 'YYYY' && fromFormat[y].length === 4) {
+        if (Number(fromFormat[y].slice(2)) < 30) {
+          toFormat[i] = `${fromFormat[y].slice(0, 2)}` + `20` + `${fromFormat[y].slice(2)}`;
+        } else {
+          toFormat[i] = `${fromFormat[y].slice(0, 2)}` + `19` + `${fromFormat[y].slice(2)}`;
+        }
+      } else if (toFormat[i] === 'YY' && fromFormat[y].length === 8) {
+        toFormat[i] = `${fromFormat[y].slice(0, 2)}` + `${fromFormat[y].slice(6)}`;
+      } else {
+        if (toFormat[i].slice(0, 1) === fromFormat[y].slice(0, 1)) {
+          toFormat[i] = fromFormat[y];
+        }
+      }
+    }
+  }
+
+  for (let i = 0; i < 3; i++) {
```

Comment:
let's simplify it. avoid one loop.

Reply:
It does not work with one loop. Please give me a hint.

Reply:
use array methods `includes`, `split`, `indexOf`.  You can split you date and check values by array indexes. 
![image](https://user-images.githubusercontent.com/70009205/204580057-f19bec14-82f1-45ea-9791-276b49b5c704.png)

Reply:
Completely new solution were added

Reply:
Solution #3 using includes, split, indexOf were added.